search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UsageMetricsFilterTransformer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.search.clients.transformers.filter;
+
+import static io.camunda.search.clients.query.SearchQueryBuilders.and;
+import static io.camunda.search.clients.query.SearchQueryBuilders.dateTimeOperations;
+import static io.camunda.search.clients.query.SearchQueryBuilders.stringTerms;
+
+import io.camunda.search.clients.query.SearchQuery;
+import io.camunda.search.filter.Operation;
+import io.camunda.search.filter.UsageMetricsFilter;
+import io.camunda.webapps.schema.descriptors.IndexDescriptor;
+import io.camunda.webapps.schema.descriptors.operate.index.MetricIndex;
+import io.camunda.webapps.schema.descriptors.tasklist.index.TasklistMetricIndex;
+import java.util.ArrayList;
+
+public class UsageMetricsFilterTransformer implements FilterTransformer<UsageMetricsFilter> {
+
+  private final TasklistMetricIndex tasklistMetricIndex;
+  private final MetricIndex operateMetricIndex;
+  private UsageMetricsFilter filter;
+
+  public UsageMetricsFilterTransformer(
+      final TasklistMetricIndex tasklistMetricIndex, final MetricIndex metricIndex) {
+    this.tasklistMetricIndex = tasklistMetricIndex;
+    operateMetricIndex = metricIndex;
+  }
+
+  @Override
+  public SearchQuery toSearchQuery(final UsageMetricsFilter filter) {
+    this.filter = filter;
+    final var queries = new ArrayList<SearchQuery>();
+    queries.add(stringTerms("event", filter.events()));
+    queries.addAll(dateTimeOperations("eventTime", Operation.gte(filter.startTime()).values()));
+    queries.addAll(dateTimeOperations("eventTime", Operation.lte(filter.endTime()).values()));
```

Comment:
🔧 Looking at other transformers, e.g. the [UserTaskFilterTransformer](https://github.com/camunda/camunda/blob/bfbfcf6a9c4ac5959ee109d72c60801e9a3fa533/search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/filter/UserTaskFilterTransformer.java), we could improve here in the following:
* Reuse the field names defined in the webapps schema classes instead of using "magic" strings.
* If an `Operation` is already defined (by the way, I think it technically could be a list of operations here already), we take the operator and value from that instead of hardcoding it here again. If we change the query in the REST API, it will still be `gte` and `lte` as defined here.

Reply:
Simplifying this by hardwiring the operations and using simple values in the filter